src/test/java/mate/academy/service.impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package mate.academy.service.impl;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceImplTest {
+    private static UserService userService;
+    private static final String email = "usermail@gmail.com";
+    private User user;
+    private User actual;
```

Comment:
you use this field in one test method, make it local. Look through other classes and make local variables from fields where it's possible